src/Microsoft.DotNet.ProjectModel/Files/IncludeFilesResolver.cs

```diff
@@ -74,28 +74,68 @@ public static IEnumerable<IncludeEntry> GetIncludeFiles(IncludeContext context,
                 }
                 else if (isFile && files.Any())
                 {
-                    includeEntries.Add(new IncludeEntry(targetBasePath, files.First()));
+                    includeEntries.Add(
+                        new IncludeEntry(
+                            targetBasePath,
+                            Path.GetFullPath(
+                                Path.Combine(sourceBasePath, PathUtility.GetPathWithDirectorySeparator(files.First().Path)))));
                 }
-                else
+                else if (!isFile)
```

Comment:
We have always supported mapping directory to a file in `packInclude` as long as the directory contains a single file. Otherwise we show an error https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.DotNet.ProjectModel/Files/IncludeFilesResolver.cs#L66.